### ISSUE TITLE
Feature request: native Write InfluxDB plugin

### ISSUE THREAD

**Issue by octo** [collectd/collectd#1352]
#696 requests a native **write_influxdb plugin** and #1319 proposes such a plugin, but needs to be simplified before it can be merged. Since there clearly is interest in such a plugin, I'd like to describe the ideal solution here and hope that someone will signal willingness to work on this. I, in turn, offer any help necessary to get this implemented and merged.

There are three steps to implement this:
 *  [InfluxDB's line protocol](https://influxdb.com/docs/v0.9/write_protocols/write_syntax.html) should be implemented in a file called `src/utils_format_influxdb.c`. Take a look at `src/utils_format_graphite.c` for a similar implementation of a line based protocol. Since InfluxDB uses labels to identify metrics, it might be interesting to look at the name mangling done by the [prometheus/collectd_exporter](https://github.com/prometheus/collectd_exporter) to get a reasonable naming schema.
 *  Add support for the format to the *write_http plugin*. Look for *format_json* and *FORMAT_JSON* in `src/write_http.c` and add an appropriate case for InfluxDB.
 *  Add a separate *write_influxdb plugin* to use the line protocol on a plain TCP socket, i.e. without HTTP. Use `src/write_graphite.c` as a starting point for this.

The [build system](https://collectd.org/wiki/index.php/Build_system) can be a scary place, but luckily it should be possible to copy and adapt the setup for the *write_graphite plugin* in large parts. There are a couple more pages in [the "Development" category](https://collectd.org/wiki/index.php/Category:Development) of collectd's wiki that may help you get started, of particular interest may be [Plugin architecture](https://collectd.org/wiki/index.php/Plugin_architecture) and [Coding style](https://collectd.org/wiki/index.php/Coding_style).

If you're interested in working on this, please ping me and/or this issue. Thank you!

**Comment by faxm0dem** [comment]
About the last point: I think the "line protocol" is just specifying the text format, but it is supposed to be sent over HTTP and not plain TCP

**Comment by supershal** [comment]
+1.
How can we support batching of the points in influxdb plugin. for example, if sending interval is "10s" then is it possible to group different measurements such as cpu, memory and disk etc grouped together in one batch and sent to influxdb with single HTTP per batch?

**Comment by nickrobinson** [comment]
@supershal It is possible to batch datapoints sent to InfluxDB by including a newline between values.

For example:
```
curl -i -XPOST 'http://localhost:8086/write?db=mydb' --data-binary 'cpu_load_short,host=server02 value=0.67
cpu_load_short,host=server02,region=us-west value=0.55 1422568543702900257
cpu_load_short,direction=in,host=server01,region=us-west value=2.0 1422568543702900257'
```

**Comment by nickrobinson** [comment]
Is this still needed since the write_tsdb plugin seems to be working fine with InfluxDB?

**Comment by wei-hai-ck** [comment]
I think we still need `write_influxdb` as `write_tsdb` doesn't support credential via HTTP and for centOS 6 there is no `collectd-tsdb` to install as dependency.

**Comment by septatrix** [comment]
I agree that a new native plugin would be great as the current `write_influxdb_udp` does not work with InfluxDB 2.x anymore. It also seems like the new version should be relatively easy to support as there is now [a HTTP InfluxDB API](https://docs.influxdata.com/influxdb/v2.1/write-data/developer-tools/api/)
